diff --git a/implicit/gpu/CMakeLists.txt b/implicit/gpu/CMakeLists.txt
index 85a2c55..70f5ffc 100644
--- a/implicit/gpu/CMakeLists.txt
+++ b/implicit/gpu/CMakeLists.txt
@@ -14,7 +14,7 @@ else()
     add_cython_target(_cuda CXX)
 
     # use rapids-cmake to install dependencies
-    file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-23.04/RAPIDS.cmake
+    file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-23.06/RAPIDS.cmake
         ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
     include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
     include(rapids-cmake)
@@ -22,9 +22,29 @@ else()
     include(rapids-cuda)
     include(rapids-export)
     include(rapids-find)
+    include(${rapids-cmake-dir}/cpm/package_override.cmake)
+
     rapids_cpm_init()
     rapids_cmake_build_type(Release)
 
+    file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/override.json
+      [=[
+        {
+          "packages" : {
+            "Thrust" : {
+              "version" : "1.17.2",
+              "git_url" : "https://github.com/benfred/thrust.git",
+              "git_tag" : "no_cmake_find_root_path",
+              "git_shallow" : true,
+              "always_download" : true,
+              "exclude_from_all" : false
+
+            }
+          }
+        }
+    ]=])
+    rapids_cpm_package_override(${CMAKE_CURRENT_BINARY_DIR}/override.json)
+
     # get rmm
     include(${rapids-cmake-dir}/cpm/rmm.cmake)
     rapids_cpm_rmm(BUILD_EXPORT_SET implicit-exports INSTALL_EXPORT_SET implicit-exports)
diff --git a/setup.py b/setup.py
index c6c6de2..fef5769 100644
--- a/setup.py
+++ b/setup.py
@@ -58,4 +58,5 @@ setup(
     install_requires=["numpy", "scipy>=0.16", "tqdm>=4.27"],
     setup_requires=setup_requires,
     cmake_process_manifest_hook=exclude_non_implicit_cmake_files,
+    cmake_args=["--trace-expand", "--debug-find-pkg=Thrust"],
 )
